src/DotVVM.Framework/Resources/Scripts/metadata/metadataHelper.ts

```diff
@@ -0,0 +1,40 @@
+﻿import { getTypeInfo } from "./typeMap";
+
+export function getTypeId(viewModel: any): string {
+    if (typeof viewModel !== "object") {
+        throw Error("Expected object but received: " + typeof viewModel);
+    }
+
+    return ko.unwrap(viewModel.$type);
+}
+
+export function getTypeMetadata(viewModel: any): TypeMetadata {
+    let typeId = getTypeId(viewModel);
+    return getTypeInfo(typeId);
+}
+
+export function getEnumMetadata(enumMetadataId: string): EnumTypeMetadata {
+    let metadata = getTypeInfo(enumMetadataId);
+    if (metadata.type !== "enum") {
+        throw new Error("Expected enum, but received object");
+    }
+
+    return metadata as EnumTypeMetadata;
+}
+
+export function getEnumValue(identifier: string | number, enumMetadataId: string): number | null {
+    let metadata = getEnumMetadata(enumMetadataId);
+    if (typeof identifier === "string") {
+        return metadata.values[identifier];
+    }
+    else {
+        // Ensure this number is not already defined
+        for (const key in metadata.values) {
+            if (metadata.values[key] === identifier) {
+                return null;
+            }
+        }
```

Comment:
I don't understand this check - why is it a problem if the number **is** defined?

Reply:
Originally, I did not like the idea of passing arbitrary numbers - especially those that have defined string identifiers. But I agree it probably does not make much sense so I took it out.